applications/dashboard/models/class.usermodel.php

```diff
@@ -2871,6 +2885,31 @@ public function addInsertFields(&$Fields) {
     }
 
     /**
+     * Record an IP address for a user.
+     *
+     * @param int $userID Unique ID of the user.
+     * @param string $IP Human-readable IP address.
+     */
+    public function saveIP($userID, $IP) {
+        $packedIP = ipEncode($IP);
+        $px = Gdn::database()->DatabasePrefix;
+        $currentDateTime = Gdn_Format::toDateTime();
+
+        $query = "insert into {$px}UserIP (UserID, IPAddress, DateInserted, DateUpdated)
+            values (:UserID, :IPAddress, :DateInserted, :DateUpdated)
+            on duplicate key update DateUpdated = :DateUpdated2";
+        $values = [
+            ':UserID' => $userID,
+            ':IPAddress' => $packedIP,
+            ':DateInserted' => $currentDateTime,
+            ':DateUpdated' => $currentDateTime,
+            ':DateUpdated2' => $currentDateTime
+        ];
+
+        Gdn::database()->query($query, $values);
```

Comment:
Let's try/catch this one for now to prevent sites that haven't done a db update from failing. You can return true/false on success/failure maybe.